app/src/main/java/de/tum/in/tumcampusapp/component/ui/transportation/widget/MVVWidgetConfigureActivity.kt

```diff
@@ -0,0 +1,194 @@
+package de.tum.`in`.tumcampusapp.component.ui.transportation.widget
+
+import android.app.Activity
+import android.appwidget.AppWidgetManager
+import android.content.Intent
+import android.os.Bundle
+import android.view.MenuItem
+import android.view.View
+import android.widget.AdapterView
+import android.widget.ArrayAdapter
+import android.widget.ListView
+import android.widget.Switch
+import de.tum.`in`.tumcampusapp.R
+import de.tum.`in`.tumcampusapp.component.other.general.RecentsDao
+import de.tum.`in`.tumcampusapp.component.other.generic.activity.ActivityForSearching
+import de.tum.`in`.tumcampusapp.component.other.generic.adapter.NoResultsAdapter
+import de.tum.`in`.tumcampusapp.component.ui.transportation.MVVStationSuggestionProvider
+import de.tum.`in`.tumcampusapp.component.ui.transportation.TransportController
+import de.tum.`in`.tumcampusapp.component.ui.transportation.model.efa.StationResult
+import de.tum.`in`.tumcampusapp.component.ui.transportation.model.efa.WidgetDepartures
+import de.tum.`in`.tumcampusapp.database.TcaDb
+import de.tum.`in`.tumcampusapp.utils.Utils
+import io.reactivex.android.schedulers.AndroidSchedulers
+import io.reactivex.disposables.CompositeDisposable
+import io.reactivex.schedulers.Schedulers
+
+class MVVWidgetConfigureActivity :
+        ActivityForSearching(R.layout.activity_mvv_widget_configure, MVVStationSuggestionProvider.AUTHORITY, 3),
+        AdapterView.OnItemClickListener {
+
+    private var appWidgetId: Int = 0
+    private lateinit var listViewResults: ListView
+    private lateinit var adapterStations: ArrayAdapter<StationResult>
+    private lateinit var recentsDao: RecentsDao
+
+    private lateinit var widgetDepartures: WidgetDepartures
+
+    private val disposable = CompositeDisposable()
+
+    public override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+
+        recentsDao = TcaDb.getInstance(this).recentsDao()
+
+        // Setup cancel button
+        supportActionBar!!.setDisplayHomeAsUpEnabled(true)
+        supportActionBar!!.setHomeAsUpIndicator(R.drawable.ic_action_cancel)
+
+        // Get appWidgetId from intent
+        val intent = intent
+        val extras = intent.extras
+        if (extras != null) {
+            appWidgetId = extras.getInt(
+                    AppWidgetManager.EXTRA_APPWIDGET_ID,
+                    AppWidgetManager.INVALID_APPWIDGET_ID)
+        }
+
+        val tm = TransportController(this)
+        this.widgetDepartures = tm.getWidget(appWidgetId)
+
+        val autoReloadSwitch = findViewById<Switch>(R.id.mvv_widget_auto_reload)
+        autoReloadSwitch.isChecked = this.widgetDepartures.autoReload
+        autoReloadSwitch.setOnCheckedChangeListener { _, checked ->
+            widgetDepartures.autoReload = checked
+        }
+        // TODO add handling for use location
```

Comment:
Was this TODO in here before you did the refactoring?

Reply:
Yes, that was there before refactoring...